lib/Conversion/TritonGPUToLLVM/DecomposeUnsupportedConversions.cpp

```diff
@@ -99,15 +99,21 @@ void decomposeBlockedToDotLayoutConversion(ModuleOp module) {
           return;
         }
       }
-
+      auto srcOrder = srcBlocked.getOrder();
+      auto rank = srcOrder.size();
+      SmallVector<unsigned> sharedOrder;
+      if (rank == 3) {
+        sharedOrder = gpu::getThreadOrder(dstEncoding);
+      } else {
+        sharedOrder = srcOrder;
+      }
```

Comment:
If you are going this route, you probably want to do it for all ranks, otherwise this heuristic would be incredibly counterintuitive.

Reply:
But then we might change the shared layout order for `rank != 3` with this change? I can also revert all changes except for the condition in `ReduceDataDuplication` to make it ampere specific. Then we have the same behavior as before #4904.

Reply:
I just find very weird the current behaviour for `rank==3`. @Jokeren thoughts?

Reply:
Actually block to dot decomposition can be deprecated soon
https://github.com/triton-lang/triton/compare/main...keren/dot-mma#diff-30fb59df648f6c4ec5db24c51ef728a8d56104e151bd52e8d482b6951c207aa7R95

Reply:
I agree the special condition is weird. Can you attach a test case for us to take a look at?

Reply:
When you run `python/test/unit/language/test_core.py::test_dot3d[1-1-64-64-64-32-32-float16-float32]` it will trigger this [assert](https://github.com/triton-lang/triton/blob/main/lib/Conversion/TritonGPUToLLVM/MemoryOpToLLVM.cpp#L27-L29). This happens when `wmma` is used for the dot.